rotkehlchen/tests/unit/test_chains_aggregator.py

```diff
@@ -26,42 +27,102 @@ def test_module_deactivation(blockchain):
         assert module_name not in blockchain.eth_modules
 
 
-def test_filter_active_evm_addresses(blockchain):
+@pytest.mark.parametrize('ethereum_accounts', [[]])
+def test_detect_evm_accounts(blockchain: 'ChainsAggregator') -> None:
+    """
+    Tests that the detection of EVM accounts activity in chains where they are not tracked yet
+    works as expected.
+    """
+    # Is a contract in ethereum mainnet and should not be added anywhere else despite it has
+    # activity in other chains
+    eth_addy_contract = make_evm_address()
 
-    contract_addy = make_evm_address()
-    all_addy = make_evm_address()
-    optimism_addy = make_evm_address()
-    avalanche_addy = make_evm_address()
+    # Is an EOA in optimism. Has activity in all chains. Should be added to optimism and avax
+    addy_eoa_1 = make_evm_address()
+
+    # Is an EOA in ethereum mainnet. Has activity only in ethereum and in optimism. Should be
+    # added to optimism and should not be added to avax
+    addy_eoa_2 = make_evm_address()
+
+    # Is an EOA that is initially already added everywhere. Has activity in all chains.
+    # Since is already added, should not be added again.
+    everywhere_addy = make_evm_address()
+
+    initial_accounts_data = []
+    addies_to_add = [
+        (SupportedBlockchain.ETHEREUM, eth_addy_contract),
+        (SupportedBlockchain.OPTIMISM, addy_eoa_1),
+        (SupportedBlockchain.ETHEREUM, addy_eoa_2),
+        (SupportedBlockchain.ETHEREUM, everywhere_addy),
+        (SupportedBlockchain.OPTIMISM, everywhere_addy),
+        (SupportedBlockchain.AVALANCHE, everywhere_addy),
```

Comment:
I was thinking to loop every evm chain here but this test will need adjustment anyway everytime we add a new evm chain